app/controllers/sms_controller.rb

```diff
@@ -44,7 +36,11 @@ def process_failure(result)
       result.to_h
     )
 
-    head :forbidden
+    if !@message.signature_valid?
```

Comment:
Maybe instead of `signature_valid?`, we could do something like this:

```
if @message.unsupported_message_type?
  head :ok
else
  head :forbidden
end
```

...since what we really want is render a 200 when the message type is unsupported, and error in all other cases.

Reply:
I like the general approach, but that method feels a little contrived to serve this particular case. I'm trying to think of a good way to address the concern but with a better method. (I think the concept of 'valid' seems clearer and more standardized.)

Reply:
I think the important things is that we not swallow an error here. Currently, this would swallow an error for missing fields in the request.

Reply:
I was misreading how this works in the message class. This looks like it works.